File: test/src/main/java/com/tunjid/androidbootstrap/test/idlingresources/FragmentGoneIdlingResource.java
package com.tunjid.androidbootstrap.test.idlingresources;

import androidx.fragment.app.FragmentManager;
import androidx.appcompat.app.AppCompatActivity;

/**
 * Idling resource that idles until a fragment has disappeared.
 * <p>
 * Created by tj.dahunsi on 4/29/17.
 */
public class FragmentGoneIdlingResource extends BaseFragmentIdlingResource {

    public FragmentGoneIdlingResource(AppCompatActivity activity, String fragmentTag, boolean unregisterSelf) {
        super(activity, fragmentTag, unregisterSelf);
    }

    public FragmentGoneIdlingResource(FragmentManager fragmentManager, String fragmentTag, boolean unregisterSelf) {
        super(fragmentManager, fragmentTag, unregisterSelf);
    }

    @Override
    public boolean isIdleNow() {
        // Wait until fragment is gone from fragment manager
        boolean isIdle = (findFragmentByTag() == null);

        return super.handleIsIdle(isIdle);
    }
}

File: test/src/main/java/com/tunjid/androidx/test/idlingresources/FragmentGoneIdlingResource.kt
package com.tunjid.androidx.test.idlingresources

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

/**
 * Idling resource that idles until a fragment has disappeared.
 *
 *
 * Created by tj.dahunsi on 4/29/17.
 */
class FragmentGoneIdlingResource(
        fragmentManager: FragmentManager,
        private val fragmentTag: String
) : BaseFragmentIdlingResource(fragmentManager, fragmentTag) {

    override fun onFragmentDestroyed(fm: FragmentManager, f: Fragment) {
        if (f.tag == fragmentTag) idle = true
    }
}
